nemo/collections/tts/modules/spectrogram_enhancer.py

```diff
@@ -0,0 +1,355 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# MIT License
+#
+# Copyright (c) 2020 Phil Wang
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# The following is largely based on code from https://github.com/lucidrains/stylegan2-pytorch
+
+import math
+from functools import partial
+from math import log2
+from typing import List
+
+import torch
```

Comment:
Needs to be taken care of.

Reply:
This means when you are importing `import torch` you should call all functionalities of torch with full name like use use `torch.Tensor` and `torch.nn` instead of `from torch import Tensor, nn`

Reply:
> you should call all functionalities of torch with full name

Is is a NeMo code style requirement? I see nothing wrong with it, my preference would be is "ignore / won't fix".

Reply:
I am just thinking things up on the spot right now. I am not an expert with PyTorch conventions, but I have never seen such layout/structure/usage with any other library.

I would default to just importing those 2 and  providing full paths beyond that, for clarity. The usual alternative, as mentioned in the code style warning, is to import all methods `from torch import Tensor, nn, randn, ...` but that sounds like a bad idea because so many torch functions like randn are the exact same name as equivalent numpy methods.

I see no consistency on this in the rest of NeMo codebase. It probably does not matter.

Reply:
Done

Left only `import torch` and `import torch.nn.functional as F`